lib/infield/core_ext.rb

```diff
@@ -10,5 +10,24 @@ def warn(*messages, **xargs)
     end
   end
 
-  Kernel.prepend(Core)
+  module InfieldWarningCapture
+    if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new("3.0")
+      # Ruby 3.0+ supports warn(msg, category: nil, **kwargs)
+      def warn(msg, category: nil, **kwargs)
+        super
+
+        Infield::DeprecationWarning.log(msg, callstack: caller_locations, validated: category == :deprecated)
```

Comment:
Should we just only call our `Infield::DeprecationWarning` when category is deprecated? We don't need to send a bunch of unnecessary messages if we can filter out which ones are deprecations. Otherwise LGTM